kube-runtime/src/reflector/store.rs

```diff
@@ -56,6 +69,12 @@ where
         }
     }
 
+    pub fn subscribe(&self) -> Option<ReflectHandle<K>> {
+        self.dispatcher
+            .as_ref()
+            .map(|dispatcher| dispatcher.subscribe(self.as_reader()))
+    }
```

Comment:
i feel like there's a way to avoid this unwrap on the user side with types like if we split the logic up here a bit more;

- `store_shared` returns a type that has `subscribe` on it
- `store` returns a type without

but it's a bit harder when `Writer` is taken as an arg and we would need to traitify it :thinking: